shell/platform/android/io/flutter/embedding/engine/android/FlutterSurfaceView.java

```diff
@@ -0,0 +1,159 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.android;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.SurfaceHolder;
+import android.view.SurfaceView;
+
+import java.nio.ByteBuffer;
+
+import io.flutter.embedding.engine.renderer.FlutterRenderer;
+
+/**
+ * Paints a Flutter UI on a {@link android.view.Surface}.
+ *
+ * Consider using a {@link FlutterView}, which utilizes a {@code FlutterSurfaceView}
```

Comment:
What's the use case for instantiating this or the TextureView equivalent instead of the `FlutterView` wrapper? When would somebody want to use the wrapper over these or vice versa?

Reply:
Done